File: mp3gain2026-net10/DarkToolStripRenderer.cs
using System.Drawing;
using System.Drawing.Drawing2D;
using System.Windows.Forms;

namespace Mp3Gain2026;

internal class DarkToolStripRenderer : ToolStripProfessionalRenderer
{
	private static readonly Color BgToolbar = Color.FromArgb(22, 22, 32);

	private static readonly Color BgHover = Color.FromArgb(45, 45, 65);

	private static readonly Color BgPressed = Color.FromArgb(60, 60, 85);

	private static readonly Color Accent = Color.FromArgb(99, 102, 241);

	private static readonly Color BorderClr = Color.FromArgb(55, 55, 75);

	private static readonly Color TextClr = Color.FromArgb(237, 237, 245);

	private static readonly Color SepColor = Color.FromArgb(50, 50, 70);

	protected override void OnRenderToolStripBackground(ToolStripRenderEventArgs e)
	{
		e.Graphics.Clear(BgToolbar);
	}

	protected override void OnRenderButtonBackground(ToolStripItemRenderEventArgs e)
	{
		Rectangle bounds = new Rectangle(Point.Empty, e.Item.Size);
		Color color;
		if (e.Item.Pressed)
		{
			color = BgPressed;
		}
		else
		{
			if (!e.Item.Selected)
			{
				return;
			}
			color = BgHover;
		}
		using SolidBrush brush = new SolidBrush(color);
		using GraphicsPath path = RoundedRect(bounds, 6);
		e.Graphics.SmoothingMode = SmoothingMode.AntiAlias;
		e.Graphics.FillPath(brush, path);
	}

	protected override void OnRenderSeparator(ToolStripSeparatorRenderEventArgs e)
	{
		if (e.Item.IsOnDropDown)
		{
			Rectangle rect = new Rectangle(0, 0, e.Item.Width, e.Item.Height);
			using SolidBrush brush = new SolidBrush(Color.FromArgb(40, 40, 40));
			e.Graphics.FillRectangle(brush, rect);
			using Pen pen = new Pen(SepColor);
			e.Graphics.DrawLine(pen, 30, e.Item.Height / 2, e.Item.Width - 10, e.Item.Height / 2);
			return;
		}
		int num = e.Item.Width / 2;
		using Pen pen2 = new Pen(SepColor);
		e.Graphics.DrawLine(pen2, num, 4, num, e.Item.Height - 4);
	}

	protected override void OnRenderMenuItemBackground(ToolStripItemRenderEventArgs e)
	{
		Rectangle rect = new Rectangle(Point.Empty, e.Item.Size);
		if (e.Item.IsOnDropDown)
		{
			Color color = (e.Item.Selected ? BgHover : Color.FromArgb(40, 40, 40));
			using SolidBrush brush = new SolidBrush(color);
			e.Graphics.FillRectangle(brush, rect);
			return;
		}
		if (e.Item.Pressed)
		{
			using (SolidBrush brush2 = new SolidBrush(BgPressed))
			{
				e.Graphics.FillRectangle(brush2, rect);
				return;
			}
		}
		if (e.Item.Selected)
		{
			using (SolidBrush brush3 = new SolidBrush(BgHover))
			{
				e.Graphics.FillRectangle(brush3, rect);
				return;
			}
		}
		using SolidBrush brush4 = new SolidBrush(BgToolbar);
		e.Graphics.FillRectangle(brush4, rect);
	}

	protected override void OnRenderItemText(ToolStripItemTextRenderEventArgs e)
	{
		e.TextColor = (e.Item.IsOnDropDown ? Color.WhiteSmoke : TextClr);
		base.OnRenderItemText(e);
	}

	protected override void OnRenderToolStripBorder(ToolStripRenderEventArgs e)
	{
		if (e.ToolStrip.IsDropDown)
		{
			using (Pen pen = new Pen(BorderClr))
			{
				e.Graphics.DrawRectangle(pen, new Rectangle(0, 0, e.ToolStrip.Width - 1, e.ToolStrip.Height - 1));
			}
		}
	}

	private static GraphicsPath RoundedRect(Rectangle bounds, int radius)
	{
		GraphicsPath graphicsPath = new GraphicsPath();
		int num = radius * 2;
		Rectangle rect = new Rectangle(bounds.Location, new Size(num, num));
		graphicsPath.AddArc(rect, 180f, 90f);
		rect.X = bounds.Right - num;
		graphicsPath.AddArc(rect, 270f, 90f);
		rect.Y = bounds.Bottom - num;
		graphicsPath.AddArc(rect, 0f, 90f);
		rect.X = bounds.Left;
		graphicsPath.AddArc(rect, 90f, 90f);
		graphicsPath.CloseFigure();
		return graphicsPath;
	}
}

File: mp3gain2026-net48/mp3gain2026/DarkToolStripRenderer.vb
Imports System.Drawing
Imports System.Drawing.Drawing2D
Imports System.Windows.Forms

Friend Class DarkToolStripRenderer
    Inherits ToolStripProfessionalRenderer

    Private Shared ReadOnly BgToolbar As Color = Color.FromArgb(22, 22, 32)
    Private Shared ReadOnly BgHover As Color = Color.FromArgb(45, 45, 65)
    Private Shared ReadOnly BgPressed As Color = Color.FromArgb(60, 60, 85)
    Private Shared ReadOnly Accent As Color = Color.FromArgb(99, 102, 241)
    Private Shared ReadOnly BorderClr As Color = Color.FromArgb(55, 55, 75)
    Private Shared ReadOnly TextClr As Color = Color.FromArgb(237, 237, 245)
    Private Shared ReadOnly SepColor As Color = Color.FromArgb(50, 50, 70)

    Protected Overrides Sub OnRenderToolStripBackground(e As ToolStripRenderEventArgs)
        e.Graphics.Clear(BgToolbar)
    End Sub

    Protected Overrides Sub OnRenderButtonBackground(e As ToolStripItemRenderEventArgs)
        Dim bounds As New Rectangle(Point.Empty, e.Item.Size)
        Dim bg As Color

        If e.Item.Pressed Then
            bg = BgPressed
        ElseIf e.Item.Selected Then
            bg = BgHover
        Else
            Return ' transparent
        End If

        Using brush As New SolidBrush(bg)
            Using path = RoundedRect(bounds, 6)
                e.Graphics.SmoothingMode = SmoothingMode.AntiAlias
                e.Graphics.FillPath(brush, path)
            End Using
        End Using
    End Sub

    Protected Overrides Sub OnRenderSeparator(e As ToolStripSeparatorRenderEventArgs)
        If e.Item.IsOnDropDown Then
            Dim rect As New Rectangle(0, 0, e.Item.Width, e.Item.Height)
            Using bgBrush As New SolidBrush(Color.FromArgb(40, 40, 40))
                e.Graphics.FillRectangle(bgBrush, rect)
            End Using
            Using pen As New Pen(SepColor)
                e.Graphics.DrawLine(pen, 30, e.Item.Height \ 2, e.Item.Width - 10, e.Item.Height \ 2)
            End Using
        Else
            Dim x As Integer = e.Item.Width \ 2
            Using pen As New Pen(SepColor)
                e.Graphics.DrawLine(pen, x, 4, x, e.Item.Height - 4)
            End Using
        End If
    End Sub

    Protected Overrides Sub OnRenderMenuItemBackground(e As ToolStripItemRenderEventArgs)
        Dim rect As New Rectangle(Point.Empty, e.Item.Size)
        If e.Item.IsOnDropDown Then
            Dim bg As Color = If(e.Item.Selected, BgHover, Color.FromArgb(40, 40, 40))
            Using brush As New SolidBrush(bg)
                e.Graphics.FillRectangle(brush, rect)
            End Using
        Else
            If e.Item.Pressed Then
                Using brush As New SolidBrush(BgPressed)
                    e.Graphics.FillRectangle(brush, rect)
                End Using
            ElseIf e.Item.Selected Then
                Using brush As New SolidBrush(BgHover)
                    e.Graphics.FillRectangle(brush, rect)
                End Using
            Else
                Using brush As New SolidBrush(BgToolbar)
                    e.Graphics.FillRectangle(brush, rect)
                End Using
            End If
        End If
    End Sub

    Protected Overrides Sub OnRenderItemText(e As ToolStripItemTextRenderEventArgs)
        e.TextColor = If(e.Item.IsOnDropDown, Color.WhiteSmoke, TextClr)
        MyBase.OnRenderItemText(e)
    End Sub

    Protected Overrides Sub OnRenderToolStripBorder(e As ToolStripRenderEventArgs)
        If e.ToolStrip.IsDropDown Then
            Using pen As New Pen(BorderClr)
                e.Graphics.DrawRectangle(pen, New Rectangle(0, 0, e.ToolStrip.Width - 1, e.ToolStrip.Height - 1))
            End Using
        End If
    End Sub

    Private Shared Function RoundedRect(bounds As Rectangle, radius As Integer) As GraphicsPath
        Dim path As New GraphicsPath()
        Dim d As Integer = radius * 2
        Dim arc As New Rectangle(bounds.Location, New Size(d, d))

        path.AddArc(arc, 180, 90)
        arc.X = bounds.Right - d
        path.AddArc(arc, 270, 90)
        arc.Y = bounds.Bottom - d
        path.AddArc(arc, 0, 90)
        arc.X = bounds.Left
        path.AddArc(arc, 90, 90)
        path.CloseFigure()
        Return path
    End Function

End Class
